src/System.CodeDom/src/Microsoft/VisualBasic/VBCodeGenerator.cs

```diff
@@ -0,0 +1,2931 @@
+            },
+            new string[] { // 12 characters
+                "mustoverride",
+            },
+            new string [] { // 13 characters
+                "removehandler",
+            },
+            // class_finalize and class_initialize are not keywords anymore,
+            // but it will be nice to escape them to avoid warning
+            new string [] { // 14 characters
+                "class_finalize",
+                "notinheritable",
+                "notoverridable",
+            },
+            null,           // 15 characters
+            new string [] {
+                "class_initialize",
+            }
+        };
+
+        internal VBCodeGenerator()
+        {
+        }
+
+        internal VBCodeGenerator(IDictionary<string, string> providerOptions)
+        {
+            _provOptions = providerOptions;
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Gets or the file extension to use for source files.
+        ///    </para>
+        /// </devdoc>
+        protected override string FileExtension { get { return ".vb"; } }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Gets the name of the compiler exe
+        ///    </para>
+        /// </devdoc>
+        protected override string CompilerName { get { return "vbc.exe"; } }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Tells whether or not the current class should be generated as a module
+        ///    </para>
+        /// </devdoc>
+        private bool IsCurrentModule
+        {
+            get
+            {
+                return (IsCurrentClass && GetUserData(CurrentClass, "Module", false));
+            }
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Gets the token that is used to represent <see langword='null'/>.
+        ///    </para>
+        /// </devdoc>
+        protected override string NullToken
+        {
+            get
+            {
+                return "Nothing";
+            }
+        }
+
+        private void EnsureInDoubleQuotes(ref bool fInDoubleQuotes, StringBuilder b)
+        {
+            if (fInDoubleQuotes) return;
+            b.Append("&\"");
+            fInDoubleQuotes = true;
+        }
+
+        private void EnsureNotInDoubleQuotes(ref bool fInDoubleQuotes, StringBuilder b)
+        {
+            if (!fInDoubleQuotes) return;
+            b.Append("\"");
+            fInDoubleQuotes = false;
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Provides conversion to formatting with escape codes.
+        ///    </para>
+        /// </devdoc>
+        protected override string QuoteSnippetString(string value)
+        {
+            StringBuilder b = new StringBuilder(value.Length + 5);
+
+            bool fInDoubleQuotes = true;
+            Indentation indentObj = new Indentation((ExposedTabStringIndentedTextWriter)Output, Indent + 1);
+
+            b.Append("\"");
```

Comment:
Changed all relevant single-character strings to be chars, both with StringBuilder.Append and TextWriter.Output.Write{Line}